web/vtadmin/src/components/routes/tablets/InfoDropdown.tsx

```diff
@@ -1,25 +1,58 @@
 import React, { useState } from 'react';
+import { usePingTablet, useRefreshState } from '../../../hooks/api';
 import Dropdown from '../../dropdown/Dropdown';
 import MenuItem from '../../dropdown/MenuItem';
 import { Icons } from '../../Icon';
-import PingDialog from './PingDialog';
-
+import InfoDialog, { BaseInfoDialogProps } from './InfoDialog';
 interface InfoDropdownProps {
     alias: string;
     clusterID?: string;
 }
 
 const InfoDropdown: React.FC<InfoDropdownProps> = ({ alias, clusterID }) => {
-    const [isPingOpen, setPingOpen] = useState(false);
+    const [currentDialog, setCurrentDialog] = useState('');
+    const [isOpen, setIsOpen] = useState(false);
+    const closeDialog = () => setIsOpen(false);
+
+    const openDialog = (key: string) => {
+        setCurrentDialog(key);
+        setIsOpen(true);
+    };
+
+    const dialogConfigs: Record<string, BaseInfoDialogProps> = {
+        ping: {
+            useHook: usePingTablet,
+            params: { alias, clusterID },
+            options: { enabled: false },
+            successDescription: `Successfully reached tablet ${alias} via RPC.`,
+            errorDescription: `There was an issue pinging tablet ${alias}`,
+            loadingTitle: `Pinging tablet ${alias}`,
+            loadingDescription: 'Checking to see if tablet is reachable via RPC...',
+        },
+        refresh: {
+            useHook: useRefreshState,
+            params: { alias, clusterID },
+            options: { enabled: false },
+            successDescription: `Successfully refreshed tablet ${alias}.`,
+            errorDescription: `There was an issue refreshing tablet`,
+            loadingTitle: `Refreshing tablet ${alias}`,
+            loadingDescription: 'Refreshing tablet record on tablet...',
+        },
+    };
+
+    // Default config is needed to avoid "hook order" inconsistency
+    // Queries are not executed until dialog is open; no unneccessary queries are executed as a result.
+
+    const defaultConfig = dialogConfigs['ping'];
```

Comment:
Ohh could you explain what hook order inconsistency is?

Reply:
Oh yes, I just didn't have the right words for it 😅  You've probably already seen it by its proper term haha: https://paper.dropbox.com/doc/SmileML-Component-Library--BXwJQFtEB0Qh1K6kBfjEScbnAg-8JvkIG9RugUPzG7xJzeIW Just that they have to be called in the same order. Originally I had a placeholder hook as a default config but that made the compiler unhappy when switching to a "real" hook.

Reply:
Ohhh gotcha. Thanks! 

I wonder if conditionally rendering the InfoDialog would get around it & additionally avoid the extra caution around queries firing? 

```jsx
{currentDialog && <InfoDialog isOpen={isOpen} onClose={closeDialog} {...dialogConfigs[currentDialog]} />}
```

Reply:
yes actually, it would! Though I didn't do it because then it messes with the opening/closing animations of the dialog (where it abruptly disappears/appears). We could add a delay animation but I thought this would suffice for now.

Reply:
Ohhh you're totally right! Makes sense!